### ISSUE TITLE
Pre-installing requirements for building Inky

### ISSUE THREAD

**Issue by jeremydouglass** [inkle/inky#272]
I am not an experienced node developer -- but I'm familiar with e.g. "requirements.txt" in Python, which allow you to automatically install dependencies.

When building Inky with BUILD_PACKAGE after installing node/electron on a new machine, BUILD_PACKAGE failed with:

> electron-packager: command not found
> appdmg: command not found

so:

npm install -g electron-packager
npm install -g appdmg.

This fixed the problem, but I wonder if there is already a setup script that is the normal way of indicating that these are required, or if there is a typical node way of doing this that could be added. Or documentation.

Could I have missed clicking "INSTALL_AND_RUN" the first time, and would that have installed them automatically?

**Comment by joethephish** [comment]
Yes good point! We should add those to the build script. (Hopefully it fails fast if they’re already installs and doesn’t attempt to install twice!)

**Comment by joethephish** [comment]
Oh and no, they shouldn’t be added to the INSTALL_AND_RUN script since that’s just for running a dev build of Inky, not for building for distribution.

**Comment by jeremydouglass** [comment]
Got it, thanks.

Possibly relevant?
https://docs.npmjs.com/specifying-dependencies-and-devdependencies-in-a-package-json-file

**Comment by joethephish** [comment]
I wouldn’t really say it’s a dependency for the project in general, since you don’t need to be able to package & distribute in order to do development on it. Ideally when you run the build script it should ask if you’d like to install those packages (so it doesn’t go off installing global stuff without asking).

**Comment by jeremydouglass** [comment]
I've tried building inky on macOS from master 9a3e9dc using both npm 6.9.0 and npm 6.14.5.

Inky builds successfully in both cases, but the right-hand "play" pane is empty -- it does nothing, and the arrows do nothing, whether the sample text or loading The Intercept or anything else.

Are there any particular npm requirements for building? If not, perhaps I should open a separate issue for this in case it is just a problem with master on 9a3e9dc?

**Comment by jeremydouglass** [comment]
Can anyone who is building inky give me a hint about which npm version they are using?

I am unable to build it from master without a broken play pane that never loads contents.

**Comment by jerrytron** [comment]
@jeremydouglass I also noticed that the play pane just wouldn't work while running from source. _However_ when I actually created a full distributable build, the play pane _does_ work. Just thought that would be useful to know.
